libmamba/src/api/info.cpp

```diff
@@ -86,7 +89,8 @@ namespace mamba
 
         void print_info(ChannelContext& channel_context, const Configuration& config)
         {
-            auto& ctx = Context::instance();
+            assert(&channel_context.context() == &config.context());
```

Comment:
I think it's safer to just replace all the asserts checking all former singletons with if & throw...

Reply:
This assertion should only fail if we made a mistake in the implementation of the library, user code cannot make this fail (from what I searched) so I prefer an assertion in these kind of situations (project's devs error, not user error). The issue is that debug builds are not tested if my memory is correct.

Reply:
Yeah I totally agree (I suggested replacing the asserts because we never test debug mode... but yeah maybe we just should).

Reply:
Agreed on the asserts here, let's add a debug build in a dedicated PR once this one is merged.

Reply:
I think I started the work in one of my previous branches, but I hit an issue... anyway ok let's focus on that next.